tests/e2e/wp/create.spec.ts

```diff
@@ -1,75 +1,53 @@
-  });
+  await stateHomePage.wpButton.click();
+  await expect(wpDashboard.page).toHaveURL(wpDashboard.path);
 
+  await wpDashboard.isReady();
   // check if work plans exist already or not
-  if (await startMFPWorkPlanButton.isVisible()) {
-    await startMFPWorkPlanButton.click();
-    const modal = page.getByRole("dialog");
-    await expect(modal).toBeVisible();
-    await expect(modal).toContainText("Add new MFP Work Plan");
+  if (await wpDashboard.createButton.isVisible()) {
+    await wpDashboard.createButton.click();
+    await expect(wpDashboard.modal).toBeVisible();
+    await expect(wpDashboard.modal).toContainText("Add new MFP Work Plan");
 
-    await page.getByLabel(`${currentYear}`).click();
-    await page
-      .getByLabel(`First reporting period (January 1 - June 30)`)
-      .click();
-    await page.getByRole("button", { name: "Start new" }).click();
-    // Confirm created WP is in table
-    await expect(page.getByRole("table")).toBeVisible();
+    await wpDashboard.createNewWP();
 
-    const row = page.getByRole("row", { name: "Not Started" });
-    await expect(
-      row.getByRole("gridcell", {
-        name: `${stateName} MFP Work Plan ${currentYear} - Period ${firstPeriod}`,
-      })
-    ).toBeVisible();
-
-    const editButton = row.getByRole("button", { name: "Edit", exact: true });
-    await expect(editButton).toBeVisible();
+    // Confirm created WP is in table
+    await wpDashboard.getReports();
+    await expect(wpDashboard.firstReport).toBeVisible();
+    const editNewWPButton = wpDashboard.firstReport.getByRole("button", {
+      name: "Edit",
+      exact: true,
+    });
+    await expect(editNewWPButton).toBeVisible();
   } else {
-    await continueMFPWorkPlanButton.click();
-    const modal = page.getByRole("dialog");
-    await expect(modal).toBeVisible();
-    await expect(modal).toContainText("Continue");
+    await wpDashboard.copyoverButton.click();
+    await expect(wpDashboard.modal).toBeVisible();
+    await expect(wpDashboard.modal).toContainText("Continue");
     await page
       .getByRole("button", { name: "Continue from previous period" })
       .click();
-    // Confirm created WP is in table
-    await expect(page.getByRole("table")).toBeVisible();
 
-    const row = page.getByRole("row", { name: "Not Started" });
-    await expect(
-      row.getByRole("gridcell", {
-        name: `${stateName} MFP Work Plan ${currentYear} - Period ${secondPeriod}`,
-      })
-    ).toBeVisible();
-
-    const editButton = row.getByRole("button", { name: "Edit", exact: true });
-    await expect(editButton).toBeVisible();
+    // Confirm created WP is in table
+    await wpDashboard.getReports();
+    await expect(wpDashboard.copiedReport).toBeVisible();
+    const editCopiedReportButton = wpDashboard.copiedReport.getByRole(
+      "button",
+      {
+        name: "Edit",
+        exact: true,
+      }
+    );
+    await expect(editCopiedReportButton).toBeVisible();
```

Comment:
this logic could be moved out of the if/else because it's the same for both, right?

Reply:
It's not exactly 😬  The locator for copied report is looking for a table row with different text in it, and the edit button is within the context of that row.

Reply:
But perhaps there is a way to abstract this into a function in the page object.